SCons/Tool/ninja/__init__.py

```diff
@@ -186,7 +186,7 @@ def generate(env):
     env.Append(BUILDERS={"Ninja": ninja_builder_obj})
 
     env["NINJA_ALIAS_NAME"] = env.get("NINJA_ALIAS_NAME", "generate-ninja")
-    env['NINJA_DIR'] = env.get("NINJA_DIR", env.Dir(".ninja").path)
+    env['NINJA_DIR'] = env.get("NINJA_DIR", os.path.join(env.Dir('#').path, ".ninja"))
```

Comment:
Why os.path.join?
Just use '#/.ninja'

Reply:
because the var is fed into ninja and it doesn't know about #.
I just made it convert to path before feeding to ninja